### PR TITLE
Add D4 parameters for RSHs

### PR DESCRIPTION
**Pull request by marvinfriede**
Add parameters for the following range-separated hybrid functionals:

- CAM-QTP(01)
- LC-&omega;PBE
- LC-&omega;PBEh
- &omega;B97M-rev
- &omega;B97X-3c
- &omega;B97X-rev (from [&omega;B97X-3c paper](https://doi.org/10.1063/5.0133026))

### CODE REVIEW

File: assets/parameters.toml [hunk 1]
```diff
@@ -157,9 +162,18 @@ d4.bj-eeq-atm = { s8=2.59538499, a1=0.42088944, a2=3.28193223, doi="10.1063/1.50
 d4.bj-eeq-mbd = { s8=2.36461524, a1=0.41518379, a2=3.19365471, doi="10.1063/1.5090222" }
 
 [parameter.lcblyp]
+reference.doi = ["10.1063/1.1688752"]
 d4.bj-eeq-atm = { s8=1.60344180, a1=0.45769839, a2=7.86924893, doi="10.1063/1.5090222" }
 d4.bj-eeq-mbd = { s8=2.40109962, a1=0.47867438, a2=8.01038424, doi="10.1063/1.5090222" }
 
+[parameter.lcwpbe]
+reference.doi = ["10.1063/1.2954017", "10.1021/ct800530u"]
+d4.bj-eeq-atm = { s8=1.170, a1=0.378, a2=4.816, doi="" } MISSING
```
Comment (awvwgk) [inline assets/parameters.toml:171]:
```suggestion
d4.bj-eeq-atm = { s8=1.170, a1=0.378, a2=4.816 }  # TODO: Add DOI
```

File: assets/parameters.toml [hunk 2]
```diff
@@ -157,9 +162,18 @@ d4.bj-eeq-atm = { s8=2.59538499, a1=0.42088944, a2=3.28193223, doi="10.1063/1.50
...
+
+[parameter.lcwpbeh]
+reference.doi = ["10.1063/1.3073302"]
+d4.bj-eeq-atm = { s8=1.318, a1=0.386, a2=5.010, doi="" } MISSING
```
Comment (awvwgk) [inline assets/parameters.toml:175]:
```suggestion
d4.bj-eeq-atm = { s8=1.318, a1=0.386, a2=5.010 }  # TODO: Add DOI
```

File: assets/parameters.toml [hunk 3]
```diff
@@ -397,6 +423,10 @@ d4.bj-eeq-atm = { s8=0.6633, a1=0.4288, a2=3.9935, doi="10.1002/jcc.26411" }
 reference.doi = ["10.1063/1.4952647"]
 d4.bj-eeq-atm = { s8=0.7761, a1=0.7514, a2=2.7099, doi="10.1002/jcc.26411" }
 
+[parameter.wb97m-rev]
+reference.doi = ["10.1063/1.4952647"]
+d4.bj-eeq-atm = { s8=0.842, a1=0.359, a2=4.668 , doi="" } MISSING
```
Comment (awvwgk) [inline assets/parameters.toml:428]:
```suggestion
d4.bj-eeq-atm = { s8=0.842, a1=0.359, a2=4.668 }  # TODO: Add DOI
```

**Review by awvwgk — CHANGES_REQUESTED** [review]
Could you also add tests for the newly added parameters?

File: app/driver.f90 [hunk 4]
```diff
@@ -116,12 +118,30 @@ subroutine run_main(config, error)
...
+   gc = config%gc
    if (config%mbdscale) s9 = config%inp%s9
    if (config%rational) then
       if (config%has_param) then
          param = config%inp
       else
-         call get_rational_damping(config%method, param, s9)
+         is = index(config%method, '/')
+         if (is == 0) is = len_trim(config%method) + 1
+         functional = lowercase(config%method(:is-1))
```
Comment (awvwgk) [inline app/driver.f90:130]: Can we get the functional ID instead and compare this instead of the string?

Comment (marvinfriede) [reply app/driver.f90:130]: We could, but then we have to take care to never change the order in the ID enumerator.

Comment (awvwgk) [reply app/driver.f90:130]: That would be fine just need to be internally consistent.

File: app/driver.f90 [hunk 5]
```diff
@@ -116,12 +118,30 @@ subroutine run_main(config, error)
...
+         id = get_functional_id(functional)
+
+         ! special case: r2SCAN-3c (modifies s9, ga, gc)
+         if (id == 102) then
```
Comment (awvwgk) [inline app/driver.f90:134]: I would either make the r2SCAN-3c ID public or define a function to perform this check in the param module.

Comment (marvinfriede) [reply app/driver.f90]: I made the r2SCAN-3c ID public.

**Review by awvwgk — APPROVED** [review]

**Pull request merged**